docs/environments.md

```diff
@@ -0,0 +1,308 @@
+# Environments
+
+Environments is a feature that allows you to deploy the same project to multiple places under multiple names. 
+
+## Usage
+
+A common use case that this feature enables is deploying your worker to a staging subdomain before pushing your worker to production. The top-level configuration will be used when running `wrangler publish`. In addition, you can specify multiple environments that you would like to deploy code to. Each environment can inherit properties from the configuration at the top level. `type` will always be inherited from the top-level configuration, you cannot specify different types for different environments. `name` is inherited and modified if left out of the environment configuration, a worker named `my-worker` with an environment `[env.dev]` would become `my-worker-dev`. Fields that can be inherited from the top level configuration are `account_id`, `zone_id`, `workers_dot_dev`, and `webpack_config`. `kv_namespaces` and `route` must be defined for each environment and will not be inherited.
+
+### Concepts
```

Comment:
I suggest a tiny restructure:
\# Environments
.. include overview of usage here
~~## Usage~~
~~## Concepts~~ Top Level Config
... include caveats on inheritance here